ledger/src/blockstore.rs

```diff
@@ -1921,6 +1922,127 @@ impl Blockstore {
             .map(|signatures| signatures.iter().map(|(_, signature)| *signature).collect())
     }
 
+    pub fn get_confirmed_signatures_for_address2(
+        &self,
+        address: Pubkey,
+        highest_confirmed_root: Slot,
+        start_after: Option<Signature>,
+        limit: usize,
+    ) -> Result<Vec<ConfirmedTransactionStatusWithSignature>> {
+        datapoint_info!(
+            "blockstore-rpc-api",
+            (
+                "method",
+                "get_confirmed_signatures_for_address2".to_string(),
+                String
+            )
+        );
+
+        let (mut slot, mut start_after) = match start_after {
+            None => (highest_confirmed_root, None),
+            Some(start_after) => {
+                let confirmed_transaction =
+                    self.get_confirmed_transaction(start_after, Some(UiTransactionEncoding::Json))?;
+                match confirmed_transaction {
+                    None => return Ok(vec![]),
+                    Some(ConfirmedTransaction { slot, transaction }) => {
+                        // Ensure that `start_after` is from a transaction that contains `address`
```

Comment:
This limitation isn't great.  The BigTable implementation doesn't care, you can `start_after` any signature.  But we can relax this requirement later -- it's probably not too often that the user wants to start listing after an arbitrary transaction

Reply:
It doesn't seem like the blockstore implementation cares either...?
Ohhh, see my comment below about `if let Some(start_after) = start_after`.

Reply:
Fixed